services/galley/src/Galley/Cassandra/TeamFeatures.hs

```diff
@@ -140,13 +140,61 @@ setSelfDeletingMessagesStatus tid status = do
           <> ", self_deleting_messages_ttl) "
           <> "values (?, ?, ?)"
 
+setPaymentStatus ::
+  forall (a :: TeamFeatureName) m.
+  ( MonadClient m,
+    HasPaymentStatusCol a
+  ) =>
+  Proxy a ->
+  TeamId ->
+  PaymentStatus ->
+  m PaymentStatus
+setPaymentStatus _ tid (PaymentStatus paymentStatus) =
+  case paymentStatusCol @a of
+    Nothing -> pure $ PaymentStatus PaymentUnlocked -- todo: what should we do here? probably return an error, UpdatePaymentStatusNotAllowed or sth. similar
```

Comment:
```suggestion
    Nothing -> pure $ PaymentStatus PaymentUnlocked -- TODO(battermann): what should we do here? probably return an error, UpdatePaymentStatusNotAllowed or sth. similar
```

And yes, we should throw an error here.

We could avoid this runtime error by splitting up `HasPaymentStatusCol` into one for read access and one for write access.  The one for read access returns a `Maybe String`, and the one for write access is just instantiated for those features that actually do write their payment status to cassandra.  Then the type checker make sure we can't write this line in any reasonable way.